Support CircleMarker shapes

cosme-benito opened the pull request:
Added support for creating CircleMarker shapes.

cosme-benito commented:
Found a bug when removing circle markers with right click. When I toggled edit mode off it triggered a Null Pointer Exception. Looking into it now, added WIP to the title.

codeofsumit commented:
HI @cosme-benito , thank you for your work on this 👍 
Do you mind rebasing this on the latest `develop`? I've added a testing framework and maybe you could even add some tests to this PR.
I hope to get some time to manually check your addition in the next days

cosme-benito commented:
Merged with the develop branch again, added support for the toolbar (including SVG) and tried to extend the Marker classes when possible.

I'm working on the cypress tests but I'm having a hard time getting it to work. It seems cypress is trying to click on the map to add the circle markers before it's ready to accept clicks. Hopefully I can add them soon.

cosme-benito commented:
Basics tests added.

cosme-benito commented:
Merged with the develop branch again.

m1dst commented:
Just came across this when I found that my circle markers wouldn’t move.  Seems this might fix it?  If so I might have to replicate for a few other types of markers I’ve created (triangles and stars at the moment).  I might be better just merging in to a simple extended marker with options.

Is this PR in a state where it should be tested/merged?

cosme-benito commented:
Leaflet PM doesn't support CircleMarkers, that's why it doesn't work. This
code is ready for testing and it should fix your problem for CircleMarkers
in particular. Adapting this PR to other kind of markers shouldn't too
hard, extending the Circle markers classes will do most of the work.


On Fri, Apr 26, 2019, 18:35 James Patterson <notifications@github.com>
wrote:

> Just came across this when I found that my circle markers wouldn’t move.
> Seems this might fix it? If so I might have to replicate for a few other
> types of markers I’ve created (triangles and stars at the moment). I might
> be better just merging in to a simple extended marker with options.
>
> Is this PR in a state where it should be tested/merged?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/codeofsumit/leaflet.pm/pull/282#issuecomment-487139313>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACKARDOW6BU26RI2GEKGHLDPSM4O7ANCNFSM4ETH2ULA>
> .
>

codeofsumit commented:
Reviewing this PR and adding Circle Marker Support to leaflet.pm is next on my list

codeofsumit commented on src/js/Edit/L.PM.Edit.CircleMarker.js:
```diff
@@ -0,0 +1,131 @@
+    }
+
+    // // init markers
+    this._createHintMarker();
```
It seems overkill to have hint markers for circleMarkers to drag them. Please use 

```js
if (this.options.draggable) {
  this.enableLayerDrag();
}
```

to make circleMarkers draggable

codeofsumit commented on src/js/Edit/L.PM.Edit.CircleMarker.js:
```diff
@@ -0,0 +1,131 @@
+  enabled() {
+    return this._enabled;
+  },
+  enable(options) {
```
Please add draggable and snappable as default options like so:

```js
enable(options = {draggable: true, snappable: true }) { 
  // ...
}
```

cosme-benito commented:
Thanks for the feedback, I've implemented all your suggestions except snapping in edit mode.

I'm having a hard time having it to work, the marker is getting "stuck" after hitting the first snap point.

codeofsumit commented:
@cosme-benito just tried it. It's crazy difficult. I'm building it right now and will PR it to your branch.

codeofsumit commented:
I have created a Pull Request to your branch:
https://github.com/cosme-benito/leaflet.pm/pull/1

codeofsumit commented on src/js/Toolbar/L.PM.Toolbar.js:
```diff
@@ -220,6 +222,22 @@ const Toolbar = L.Class.extend({
     };
 
+    const drawCircleMarkerButton = {
+      title: 'Draw Circle Marker',
```
Oh another thing, we have recently released a way to internationalize the plugin. Please check the other buttons and add these English titles and tests to the translation file and include it via like you can see in the other buttons. That way, the community can translate the texts into other languages.

cosme-benito commented:
Thanks a lot for your help, I don't think I could have figured the problem on my own! From my testing everything appears to be working as intended.

I didn't notice the new support for translation files, that's nice. I'll create a pull request for the portuguese language soon. I added the translations for the circle marker texts to every language except russian and romanian. The text will be in english for those languages until someone more apt than me can translate it.

Also added some tests to ensure the labels are properly translated.

cosme-benito commented:
I added the NL language file that was merged into the dev branch. 

For Dutch, Romanian and Russian languages I just copied the labels for the "Markers" which I feel is better than just writing it in English. Hopefully someone will translate it in a better way.

cosme-benito commented:
Merged with dev again.

@codeofsumit Am I missing any request change?

codeofsumit commented:
@cosme-benito No you don't, sorry for not replying sooner. I got married last weekend and I'm now on honeymoon. I'll check this again when I'm back!

cosme-benito commented:
Congratulations mate!!!

On Sat, Jun 8, 2019 at 5:56 PM Sumit Kumar <notifications@github.com> wrote:

> @cosme-benito <https://github.com/cosme-benito> No you don't, sorry for
> not replying sooner. I got married last weekend and I'm now on honeymoon.
> I'll check this again when I'm back!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/codeofsumit/leaflet.pm/pull/282?email_source=notifications&email_token=ACKARDKHBOJYZDA6LTLCNSLPZPQETA5CNFSM4ETH2ULKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXHYTUA#issuecomment-500140496>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACKARDJRAHXZCQ5YPWNURI3PZPQETANCNFSM4ETH2ULA>
> .
>

m1dst commented:
Would be great to see some love on this PR and finally get it into production.

codeofsumit commented:
I agree. Everything seems to be working @cosme-benito - I will merge it now :-)

Thanks so much for your patience!!! 🙏
